### PR TITLE
Adds pf2e flags with sourceId, uuid and type to used item chat message.

### LINKED ISSUE #4185: Add item data to "Use Item" chat messages
When using consumable items, the only indication of what item has been used is in the flavor text.

image
image

The message's `item` data returns `null`. 
Would be great if the item data was passed on to the message, much like when you post the item itself to chat below.

image

### PR DESCRIPTION
**Pull request by xdy**
Fixes #4185

### CODE REVIEW

File: src/module/item/consumable/document.ts [hunk 1]
```diff
@@ -119,16 +119,29 @@ class ConsumablePF2e extends PhysicalItemPF2e {
 
             // If using this consumable creates a roll, we need to show it
             const formula = this.system.consume.value;
+            const flags = {
+                pf2e: {
+                    origin: this
+                        ? {
```
Comment (stwlam) [inline src/module/item/consumable/document.ts:125]: do you expect `this` to be undefined?

File: src/module/item/consumable/document.ts [hunk 2]
```diff
@@ -119,16 +119,29 @@ class ConsumablePF2e extends PhysicalItemPF2e {
...
+                              type: this.type as ItemType,
+                          }
+                        : null,
+                },
+            };
             if (formula) {
                 new Roll(formula).toMessage({
                     speaker: ChatMessage.getSpeaker({ actor: this.actor }),
                     flavor: content,
+                    flags: flags,
```
Comment (stwlam) [inline src/module/item/consumable/document.ts:137]:
```suggestion
                    flags,
```

File: src/module/item/consumable/document.ts [hunk 3]
```diff
@@ -119,16 +119,29 @@ class ConsumablePF2e extends PhysicalItemPF2e {
...
                 });
             } else if (this.consumableType !== "ammo") {
                 ChatMessage.create({
                     user: game.user.id,
                     speaker: ChatMessage.getSpeaker({ actor: this.actor }),
                     content,
+                    flags: flags,
```
Comment (stwlam) [inline src/module/item/consumable/document.ts:144]:
```suggestion
                    flags,
```

File: src/module/item/consumable/document.ts (hunk 1 above)
Comment (xdy) [reply src/module/item/consumable/document.ts:125]: Eh. Not really. Will fix.

**Pull request merged**